### PR TITLE
feat(color-picker-hex-input): auto apply new color after typing/pasting hex code

### PR DESCRIPTION
**Pull request by aPreciado88**
**Related Issue:** [#7057](https://github.com/Esri/calcite-design-system/issues/7057)

## Summary
Updates `color-picker-hex-input` to auto apply new color after typing or pasting valid hex value.

### CODE REVIEW

File: packages/calcite-components/src/components/color-picker/color-picker.tsx [hunk 1]
```diff
@@ -1055,12 +1057,22 @@ export class ColorPicker
...
+    this.handleChannelChange(customEvent);
+  };
+
   handleKeyDown(event: KeyboardEvent): void {
     if (event.key === "Enter") {
       event.preventDefault();
     }
   }
 
+  handleInputFocus = (event: Event): void => {
```
Comment (jcfranco) [inline packages/calcite-components/src/components/color-picker/color-picker.tsx:1071]: Selecting text is a great enhancement, but it does seem unrelated to committing on paste/input. Could we submit this separately?

File: packages/calcite-components/src/components/color-picker/color-picker.tsx [hunk 2]
```diff
@@ -1034,6 +1034,8 @@ export class ColorPicker
         numberingSystem={this.numberingSystem}
         onCalciteInputNumberChange={this.handleChannelChange}
         onCalciteInputNumberInput={this.handleChannelInput}
+        onFocus={this.handleInputFocus}
+        onInput={this.handleChannelInputChange}
```
Comment (jcfranco) [inline packages/calcite-components/src/components/color-picker/color-picker.tsx:1038]: For consistency, can you use the calcite-specific events? Also, the event name doesn't match the event (e.g., named after the `change` event, but it's listening to `input`). Applies to the hex-input changes.

**Review by jcfranco — CHANGES_REQUESTED** [review]
Looking good, @aPreciado88! 😎

**Comment by jcfranco** [comment]
One more thing, committing value on blur or enter is by design, so this should use the `feat` type. This also matches the request.

File: packages/calcite-components/src/components/color-picker/color-picker.tsx (hunk 1 above)
Comment (aPreciado88) [reply packages/calcite-components/src/components/color-picker/color-picker.tsx]: I'll create a separate issue for this.

**Review by jcfranco — CHANGES_REQUESTED** [review]
Awesome! Two more things and this should be good to merge, can you:

1. update the test suite to cover this new workflow? I believe current tests use `blur` and `Enter`.
2. revisit the PR title to better match the feature type? `fix color not auto updating` is still bug-focused.

**Comment by aPreciado88** [comment]
@jcfranco I updated the test suite and PR title. 🚀

File: packages/calcite-components/src/components/color-picker-hex-input/color-picker-hex-input.e2e.ts [hunk 3]
```diff
@@ -348,6 +348,14 @@ describe("calcite-color-picker-hex-input", () => {
           await assertTabAndEnterBehavior("", startingHex);
         });
 
+        it("commits hex chars when typing", async () => {
+          await selectText(input);
+          await page.keyboard.type("abcdef");
```
Comment (jcfranco) [inline packages/calcite-components/src/components/color-picker-hex-input/color-picker-hex-input.e2e.ts:353]: Since committing on paste/input only works for longhand hex, can you assert on "abc" not changing the value, then assert the value changed once "def" is typed?

**Review by jcfranco — APPROVED** [review]
🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨
🎨⌨️⌨️⌨️⌨️🎨⌨️🎨🎨🎨🎨🎨⌨️🎨⌨️⌨️⌨️⌨️🎨🎨⌨️⌨️⌨️🎨🎨⌨️⌨️🎨🎨⌨️🎨🎨🎨⌨️🎨⌨️⌨️⌨️⌨️🎨
🎨⌨️🎨🎨⌨️🎨⌨️🎨🎨🎨🎨🎨⌨️🎨⌨️🎨🎨🎨🎨⌨️🎨🎨🎨🎨⌨️🎨🎨⌨️🎨⌨️⌨️🎨⌨️⌨️🎨⌨️🎨🎨🎨🎨
🎨⌨️⌨️⌨️⌨️🎨⌨️🎨🎨⌨️🎨🎨⌨️🎨⌨️⌨️⌨️🎨🎨🎨⌨️⌨️🎨🎨⌨️🎨🎨⌨️🎨⌨️🎨⌨️🎨⌨️🎨⌨️⌨️⌨️🎨🎨
🎨⌨️🎨🎨⌨️🎨⌨️🎨⌨️🎨⌨️🎨⌨️🎨⌨️🎨🎨🎨🎨🎨🎨🎨⌨️🎨⌨️🎨🎨⌨️🎨⌨️🎨🎨🎨⌨️🎨⌨️🎨🎨🎨🎨
🎨⌨️🎨🎨⌨️🎨🎨⌨️🎨🎨🎨⌨️🎨🎨⌨️⌨️⌨️⌨️🎨⌨️⌨️⌨️🎨🎨🎨⌨️⌨️🎨🎨⌨️🎨🎨🎨⌨️🎨⌨️⌨️⌨️⌨️🎨
🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨🎨

Testing `paste` isn't possible due to [Puppeteer limitations](https://github.com/Esri/calcite-design-system/pull/5189#issue-1344620750), right?

File: packages/calcite-components/src/components/color-picker-hex-input/color-picker-hex-input.e2e.ts (hunk 3 above)
Comment (aPreciado88) [reply packages/calcite-components/src/components/color-picker-hex-input/color-picker-hex-input.e2e.ts]: I added assertion to only commit longhand hex values.

**Comment by aPreciado88** [comment]
@jcfranco I tried to include a test for the copy-paste scenario. But it looks like the puppeteer limitations are still present.

File: packages/calcite-components/src/components/color-picker-hex-input/color-picker-hex-input.e2e.ts [hunk 4]
```diff
@@ -348,11 +348,18 @@ describe("calcite-color-picker-hex-input", () => {
...
 
-        it("commits hex chars when typing", async () => {
+        it("commits longhand hex chars when typing", async () => {
           await selectText(input);
-          await page.keyboard.type("abcdef");
+          await page.keyboard.type("abc");
           await page.waitForChanges();
 
+          // asserting that shorthand hex won't be committed
+          expect(await input.getProperty("value")).toBe("#b33f33");
```
Comment (jcfranco) [inline packages/calcite-components/src/components/color-picker-hex-input/color-picker-hex-input.e2e.ts:357]: You could use `startingHex` here instead of the actual hex value.

Also, I _think_ the comments can be removed since the updated test name and assertions make it clear what the test entails.

Comment (aPreciado88) [reply packages/calcite-components/src/components/color-picker-hex-input/color-picker-hex-input.e2e.ts]: I updated the test to use `startingHex` and removed the comments. 🚀
